specs/dotsigstore_bundle/pb/sigstore_bundle.proto

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.bundle.v1;
+
+// https://raw.githubusercontent.com/secure-systems-lab/dsse/9c813476bd36de70a5738c72e784f123ecea16af/envelope.proto
+import "envelope.proto";
+import "sigstore_common.proto";
+import "sigstore_rekor.proto";
+
+option go_package = "github.com/sigstore/proto/bundle/v1";
+option java_package = "dev.sigstore.proto.bundle.v1";
+option java_multiple_files = true;
+option java_outer_classname = "BundleProto";
+
+// Notes on versioning.
+// The primary message ('Bundle') MUST be versioned, by populating the
+// 'media_type' field. Semver-ish (only major/minor versions) scheme MUST
+// be used. The current version as specified by this file is:
+// application/vnd.dev.sigstore.bundle+json;version=0.1
+// The semantic version is thus '0.1'.
+
+// This message holds different constructs that captures the timestamp
+// of the time the signature was generated.
+// As this message can be either empty (no timestamps), or a combination of
+// an arbitrarily number of transparency log entries and signed timestamps,
+// it is the client's responsibility to implement any required verification
+// policies.
+// When using RFC3161 timestamps together with DSSE envelopes, the message
+// imprint (message digest) MUST be computed over the payload (octets).
+message TimestampVerificationData {
+        // This is the regular inclusion promise and proof, where
+        // the timestamp is coming from the transparency log.
+        repeated dev.sigstore.rekor.v1.TransparencyLogEntry tlog_entries = 1;
```

Comment:
@haydentherapper should we consider making this a `oneOf` in case there is a new timestamp format we'd want to support in the future?

Reply:
I had the same thought at one point, but I think the `TimestampVerificationData` is actually future proof here because it includes 0 or more `tlog_entries`. We could just add a new `repeated dev.sigstore.rekor.v2.TransparencyLogEntry tlog_entries_v2` field.

Reply:
When did we move `tlog_entries` under timestamp verification? Was this a new change?

It's not strictly timestamping. The integrated time from the bundle provides time, but the rest of the bundle is used to prove a signing occurred, regardless of time.

Reply:
The proposal came 15th of Sept, and was committed ~25 days ago: https://github.com/sigstore/cosign/pull/2204#discussion_r977382897

Reply:
> I thought we had something that allowed us to specify both timestamps and tlog entries? Is the expectation that both can be set there?

Yes, that is still the case. As both fields are `repeated` they can both be 0-n. The example verifier (which is now moved to another branch) verifies both if they are set. But as there are some different ideas on what the usage of TSA is, we can hold off on the verification flow. I'll open an issue on this topic once we have the new repo in place.

Reply:
Oh ok, I think my issue is then just with naming - Instead of `TimestampVerificationData`, can we drop `Timestamp`?

Reply:
Copying in @znewman01 's comment from 15th Sept (the one I linked to) for more context.
The message was initially called `VerificationData` (as you are proposing).

> To open the bikeshed back up: maybe `SignatureTimeVerificationData`? `TimestampEvidence`? But clarify that it has to do with time

Based on this comment I switched to `TimestampVerificationData`. I understand that it's not strictly timestamp, the tlog entry gives you more. But one important feature of the tlog entry _is_ that it gives an external witness that the signature happened during the time the certificate was valid, and I _think_ many clients will rely on this to accept the signature (beyond just verifying the signature was put on the public ledger).

I'm ok with `VerificationData` too (as I had that name first :) ).

Reply:
Since @znewman01 said he likes more nested messages, what about this structure?

```
message TimestampVerificationData {
  repeated TS3161 ts;
  // In the future
  // repeated Roughtime rtchain;
}

message VerificationData {
  repeated RekorBundle b;
  TimestampVerificationData tsv;
}
```

I'd worry that naming the message something with `Timestamp` downplays the primary purpose of Rekor, to provide transparency.

Reply:
Nice @haydentherapper, I like that proposal!